src/notificationmanager.cpp

```diff
@@ -67,25 +99,31 @@ void NotificationManager::handleUpdateActiveNotifications(const QVariantList not
 
 void NotificationManager::handleUpdateNotificationGroup(const QVariantMap notificationGroupUpdate)
 {
-    QString notificationGroupId = notificationGroupUpdate.value("notification_group_id").toString();
+    const QString chatId = notificationGroupUpdate.value(CHAT_ID).toString();
+    const QString notificationGroupId = notificationGroupUpdate.value(NOTIFICATION_GROUP_ID).toString();
     LOG("Received notification group update, group ID:" << notificationGroupId);
+
     QVariantMap notificationGroup = this->notificationGroups.value(notificationGroupId).toMap();
+    notificationGroup.insert(TYPE, notificationGroupUpdate.value(TYPE));
+    notificationGroup.insert(CHAT_ID, chatId);
+    notificationGroup.insert(NOTIFICATION_GROUP_ID, notificationGroupId);
+    notificationGroup.insert(NOTIFICATION_SETTINGS_CHAT_ID, notificationGroupUpdate.value(NOTIFICATION_SETTINGS_CHAT_ID));
+    notificationGroup.insert(IS_SILENT, notificationGroupUpdate.value(IS_SILENT));
+    notificationGroup.insert(TOTAL_COUNT, notificationGroupUpdate.value(TOTAL_COUNT));
 
-    QString chatId = notificationGroupUpdate.value("chat_id").toString();
-    notificationGroup.insert("type", notificationGroupUpdate.value("type"));
-    notificationGroup.insert("chat_id", chatId);
-    notificationGroup.insert("notification_group_id", notificationGroupId);
-    notificationGroup.insert("notification_settings_chat_id", notificationGroupUpdate.value("notification_settings_chat_id"));
-    notificationGroup.insert("is_silent", notificationGroupUpdate.value("is_silent"));
-    notificationGroup.insert("total_count", notificationGroupUpdate.value("total_count"));
+    const QVariantList addedNotifications = notificationGroupUpdate.value(ADDED_NOTIFICATIONS).toList();
+    const QVariantList removedNotificationIds = notificationGroupUpdate.value(REMOVED_NOTIFICATION_IDS).toList();
 
-    QVariantMap activeNotifications = notificationGroup.value("notifications").toMap();
+    // Determine if we need a bzzzzz
+    const AppSettings::NotificationFeedback feedbackStyle = appSettings->notificationFeedback();
+    const bool needFeedback = (feedbackStyle != AppSettings::NotificationFeedbackNone &&
+        (!addedNotifications.isEmpty() || feedbackStyle == AppSettings::NotificationFeedbackAll));
```

Comment:
That seems to cause an NGF action (bzzzz) each time a notification group is updated. On my device every time, I read a message in a chat that is not muted and if there are still unread messages in unmuted chats.

Reply:
I'll take another look at it.